src/ui/theme/utils.ts

```diff
@@ -0,0 +1,190 @@
+import { BrandingAppearance } from "src/networking/responses/branding-response";
+import {
+  Colors,
+  DEFAULT_FORM_COLORS,
+  DEFAULT_INFO_COLORS,
+  FormColorsToBrandingAppearanceMapping,
+  InfoColorsToBrandingAppearanceMapping,
+} from "./colors";
+import {
+  DefaultShape,
+  PillsShape,
+  RectangularShape,
+  RoundedShape,
+  Shape,
+} from "./shapes";
+
+const hexToRGB = (color: string): [number, number, number] | null => {
+  if (color.length == 7)
+    return [
+      parseInt(color.slice(1, 3), 16),
+      parseInt(color.slice(3, 5), 16),
+      parseInt(color.slice(5, 7), 16),
+    ];
+  if (color.length == 4)
+    return [
+      parseInt(color[1], 16),
+      parseInt(color[2], 16),
+      parseInt(color[3], 16),
+    ];
+  return null;
+};
+
+const rgbToTextColors = ([r, g, b]: [number, number, number]) => {
+  const sum = r + g + b;
+  const baseColor = sum <= 3 * 128 ? "255,255,255" : "0,0,0";
+
+  return {
+    "grey-text-dark": `rgba(${baseColor},0.8)`,
+    "grey-text-light": `rgba(${baseColor},0.50)`,
+    "grey-ui-dark": `rgba(${baseColor},0.125)`,
+    "grey-ui-light": `rgba(${baseColor},0.05)`,
+  };
+};
+
+const textColorsForBackground = (
+  backgroundColors: string,
+  defaultColors: Colors,
+) => {
+  const fallbackColors = {
+    "grey-text-dark": defaultColors["grey-text-dark"],
+    "grey-text-light": defaultColors["grey-text-light"],
+    "grey-ui-dark": defaultColors["grey-ui-dark"],
+    "grey-ui-light": defaultColors["grey-ui-light"],
+  };
+
+  if (!backgroundColors) {
+    return fallbackColors;
+  }
+
+  if (backgroundColors.startsWith("#")) {
```

Comment:
We should receive always hex colors from the branding appearance, but just to be sure this function always falls back to the fallbackColors